lib/CL/pocl_llvm_wg.cc

```diff
@@ -1605,11 +1606,16 @@ int pocl_llvm_codegen(cl_device_id Device, cl_program program, void *Modp,
   pocl_cache_write_kernel_asmfile(AsmFileName, AsmStr.c_str(), AsmStr.size());
   pocl_cache_tempname(ObjFileName, OBJ_EXT, nullptr);
 
+  std::string MCPU = Device->llvm_cpu != nullptr
+                         ? (std::string("-mcpu=") + Device->llvm_cpu)
```

Comment:
Is this always -mcpu on all architectures ? at least the code in `cmake/LLVM.cmake` suggests that it's different on some architectures.  Perhaps use the variable CLANG_MARCH_FLAG ?